app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -410,3 +411,71 @@ export const filterRecordsBySearch = (records: RQAPI.Record[], searchValue: stri
 
   return records.filter((record) => matchingRecords.has(record.id));
 };
+
+export const clearActiveKeysFromSession = (keysToBeDeleted: string[]) => {
+  const activeKeys = sessionStorage.getItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, []);
+
+  if (keysToBeDeleted.length === 0) {
+    return;
+  }
+
+  const updatedActiveKeys: string[] = [];
+
+  activeKeys.forEach((key: string) => {
+    if (!keysToBeDeleted.includes(key)) updatedActiveKeys.push(key);
+  });
+
+  sessionStorage.setItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, updatedActiveKeys);
+};
+
+/**
+ * Updates the activeKeys array to include the parent IDs of the given ID.
+ *
+ * @description This method ensures that collections are expanded in the sidebar
+ * when switching tabs. It achieves this by identifying all parent IDs of the
+ * current tab's ID and adding them to the `activeKeys` array.
+ *
+ * @param records - Array of API client records.
+ * @param id - The ID of the record whose parent IDs need to be found.
+ * @param activeKeys - Array of IDs that are already active (expanded) in the sidebar.
+ *
+ * @returns The updated activeKeys array, including all parent IDs.
+ */
+export const updateActiveKeys = (records: RQAPI.Record[], id: RQAPI.Record["id"], activeKeys: RQAPI.Record["id"][]) => {
```

Comment:
Reorder the params and update the function name
```suggestion
export const updateActiveKeys = (id: RQAPI.Record["id"], activeKeys: RQAPI.Record["id"][], records: RQAPI.Record[]) => {
```